core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -260,7 +260,7 @@ private void addColumnsRecords(QualifiedTablePrefix prefix)
                         prefix.getCatalogName(),
                         tableName.getSchemaName(),
                         tableName.getTableName(),
-                        column.getName(),
+                        column.getMappedNameOrName(),
```

Comment:
Why change in JDBC connector changes the engine?

Reply:
The core problem is if we keep both the remote name and the mapped name around and decide based on context which name to use, then it requires engine changes. 

I might be able to remove the need to keep both the remote and mapped names around and simply do all the mapping in the jdbc client by modifying `metadata.getColumnHandles()` to return a `Map<String, ColumnHandle>` with the keys being mapped names instead of remote names:
https://github.com/trinodb/trino/blob/48451cbe5b324947db3c04a2225367ef56586cd0/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L464-L493

I'll explore that a bit and see what shakes out.

Reply:
Will close this PR and open another one, that approach is much cleaner thanks @kokosing